pkg/auth/dependency/interaction/flows/auth_api_sso.go

```diff
@@ -17,13 +25,17 @@ func (f *AuthAPIFlow) LoginWithOAuthProvider(oauthAuthInfo sso.AuthInfo, codeCha
 		interaction.IdentityClaimOAuthProfile:   oauthAuthInfo.ProviderRawProfile,
 		interaction.IdentityClaimOAuthClaims:    oauthAuthInfo.ProviderUserInfo.ClaimsValue(),
 	}
+	state := map[string]string{
+		AuthAPIStateOAuthCodeChallenge: codeChallenge,
+	}
 	i, err := f.Interactions.NewInteractionLogin(&interaction.IntentLogin{
 		Identity: interaction.IdentitySpec{
 			Type:   authn.IdentityTypeOAuth,
 			Claims: claims,
 		},
 	}, "")
 	if err == nil {
+		i.State = state
```

Comment:
We should not overwrite the state, but insert the code challenge into the state.